include/service/entities/site_service.js

```diff
@@ -0,0 +1,491 @@
+                cb(err, null)
+            } else if (!site) {
+                cb(new Error('Site not found'), null);
+            } else if (!site.active) {
+                cb(new Error('Site not active'), null);
+            } else {
+                pb.RequestHandler.activateSite(site);
+                cb(err, result)
+            }
+        });
+    };
+
+    /**
+     * Given a site uid, deactivate if the site exists so that user facing routes are off.
+     * @method stopAcceptingSiteTraffic
+     * @param {String} siteUid - site unique id
+     * @param {Function} cb - callback function
+     */
+    SiteService.prototype.stopAcceptingSiteTraffic = function(siteUid, cb) {
+        var dao = new pb.DAO();
+        dao.loadByValue('uid', siteUid, 'site', function(err, site) {
+            if(util.isError(err)) {
+                cb(err, null)
+            } else if (!site) {
+                cb(new Error('Site not found'), null);
+            } else if (site.active) {
+                cb(new Error('Site not deactivated'), null);
+            } else {
+                pb.RequestHandler.deactivateSite(site);
+                cb(err, result)
+            }
+        });
+    };
+
+    /**
+     * Load all sites into memory.
+     * @method initSites
+     * @param {Function} cb
+     */
+    SiteService.prototype.initSites = function(cb) {
+        if (pb.config.multisite.enabled && !pb.config.multisite.globalRoot) {
+            cb(new Error("A Global Hostname must be configured with multisite turned on."), false);
+        }
+        else {
+            this.getAllSites(function (err, results) {
+                if (err) {
+                    cb(err);
+                } else {
+                    util.forEach(results, function (site) {
+                        pb.RequestHandler.loadSite(site);
+                    });
+                    // To remain backwards compatible, hostname is siteRoot for single tenant
+                    // and active allows all routes to be hit.
+                    // When multisite, use the configured hostname for global, turn off public facing routes,
+                    // and maintain admin routes (active is false).
+                    pb.RequestHandler.loadSite({
+                        displayName: pb.SiteService.GLOBAL_SITE,
+                        uid: pb.SiteService.GLOBAL_SITE,
+                        hostname: pb.config.multisite.enabled ? url.parse(pb.config.multisite.globalRoot).host : url.parse(pb.config.siteRoot).host,
+                        active: pb.config.multisite.enabled ? false : true
+                    });
+                    cb(err, true);
+                }
+            });
+        }
+    };
+
+    // Generate a site unique id.
+    function getUid() {
+        return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function(c) {
+            var r = Math.random()*16|0,v=c=='x'?r:r&0x3|0x8;return v.toString(16);
+        });
+    }
```

Comment:
Is this a specific format you all needed or would util.uniqueId not work for you?